mtbl.pyx

```diff
@@ -120,7 +148,7 @@ cdef class iterkeys(object):
             res = mtbl_iter_next(self._instance, &key, &len_key, &val, &len_val)
         if res == mtbl_res_failure:
             raise StopIteration
-        return <bytes> key[:len_key]
+        return from_bytes(key[:len_key])
```

Comment:
I feel a bit funny with this being the default behaviour with no way to disable it. How about we add a kwargs bool to __init__ to turn it off?

Reply:
Okay I added a "return_bytes" option when creating a reader, merger, or sorter. The default is False so if you pass in 'abc' you will get back 'abc' like you did in Python2. But if you initialize your reader `return_bytes=True` you'll get back b'abc'

I added a bunch of tests for that functionality as well.